File: boss-2026/coin-selection-challenge-winterrdog/solution/cpp/main.cc
#include "./coin-selection.hh"
#include "./common.hh"

int main(void) {
  // Create wallet
  bcli("createwallet student false true");

  // Import descriptors
  // NOTE:
  // "PROJECT_DIR" is passed in as a define during compilation
  fs::path student_wallet{fs::path(PROJECT_DIR) / ".." / "datadir" /
                          "student_wallet.json"};
  std::string student_wallet_data{read_file(student_wallet)};
  std::string descriptors{remove_whitespace(student_wallet_data)};

  bcli("importdescriptors " + descriptors);

  // Import payments
  fs::path payments_file{fs::path(PROJECT_DIR) / ".." / "payments.json"};
  std::string payments_json{read_file(payments_file)};
  json payments_data{json::parse(payments_json)};
  std::vector<Payment> payments{payments_data.get<std::vector<Payment>>()};

  // Make payments
  make_payments(payments);

  return 0;
}

File: boss-2026/coin-selection-challenge-winterrdog/solution/rust/src/main.rs
use boss_coin_selection::*;

use std::fs;
use std::path::PathBuf;

fn main() {
    // Create wallet
    bcli("createwallet student false true");
    // Import descriptors
    let student_wallet = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
                         .join("..")
                         .join("..")
                         .join("datadir")
                         .join("student_wallet.json");
    let student_wallet_data = fs::read_to_string(student_wallet).unwrap();
    let descriptors: String = student_wallet_data.chars().filter(|c| !c.is_whitespace()).collect();
    bcli(&format!("importdescriptors {descriptors}"));
    // Import payments
    let payments_file = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
                        .join("..")
                        .join("..")
                        .join("payments.json");
    let payments_json = fs::read_to_string(payments_file).unwrap();
    let payments: Vec<Payment> = serde_json::from_str(&payments_json).unwrap();
    // Make payments
    make_payments(&payments);
}
